### DISCUSSION TITLE
`StoreDirectory` marked as `@internal`

### DISCUSSION THREAD

**Discussion by sirkalmi** [JaffaKetchup/flutter_map_tile_caching#76] [Q&A] [answered]
Hello dear developers!

I create a TileLayer using the method below.

```
  static TileLayer tileLayer(String urlTemplate, double maxZoom) => TileLayer(
        urlTemplate: urlTemplate,
        maxZoom: maxZoom,
        retinaMode: false,
        tileProvider: FMTCTileProvider(
          storeDirectory: StoreDirectory(FMTC.instance.rootDirectory, 'osm_bright'),
          settings: FMTCTileProviderSettings(
            behavior: CacheBehavior.cacheFirst,
          ),
        ),
      );
```

It works, but it bothers me that it complains about:
The member 'StoreDirectory' can only be used within its package.

Why is StoreDirectory internally annotated?
Shouldn't that be how I use it?

Tank you,
Mariák Kálmán

**Comment by JaffaKetchup** [comment, marked as answer]
Hi @sirkalmi,

Please make sure you have read the entire documentation. It looks like you may be trying to use FMTC like it was pre v5 - if so, you'll need to upgrade fully. If not:

`FMTCTileProvider` and `StoreDirectory` should not be accessed through their respective objects - they are marked with an `@internal` annotation to warn against this. Instead, follow both sections at https://fmtc.jaffaketchup.dev/usage/roots-and-stores.  
The purpose of this is to simplify usage to the 'chains' described on that page, whilst allowing flexibility within this project and for advanced devs that may be taking FMTC to the max.

_Note that this is similar (in the way of misunderstanding) to the original issue in discussion #29, although you spotted the 'internal' warning and he didn't._

If you need more help, please ask under this comment, and I'll be happy to assist :)

**Reply by sirkalmi** [reply]
Thank you for your help! The essential difference between square and round brackets was not clear at first.

**Reply by sirkalmi** [reply]
My method has been simplified as follows. I've tried this before, but by mistake I use round brackets, it didn't work that way.

```
  static TileLayer tileLayer(String urlTemplate, double maxZoom) => TileLayer(
        urlTemplate: urlTemplate,
        maxZoom: maxZoom,
        retinaMode: false,
        tileProvider: FMTC.instance['osm_bright'].getTileProvider(),
      );
```
